src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java

```diff
@@ -130,4 +131,8 @@ private static void playHurtSound(Player player) {
         player.playNotifySound(SoundEvents.BONE_BLOCK_BREAK, SoundSource.PLAYERS, 1f, player.getRandom().nextFloat() + 1f);
         player.playNotifySound(SoundEvents.PLAYER_HURT, SoundSource.PLAYERS, 1f, player.getRandom().nextFloat() + 1f);
     }
+
+    public static boolean isInSurvivalMode(Player player) {
+        return !player.isSpectator() && !player.isCreative();
```

Comment:
Just check for survival alone ? `player.isSurvival()`

Reply:
When I was looking around last night I couldn't for the life of me find an ``isSurvival()`` in the class where ``isCreative()``/``isSpetator()`` was located hence my making that method, why vanilla do this I don't know but this was the closest I could get.